Ignore error return codes from `update_exporter_deb.py`

simu opened the pull request:
We ignore errors when trying to push the updated packages to Prometheus, since a failure of that process shouldn't fail the maintenance job for a node.


Additionally, we catch exceptions thrown by `pushadd_to_gateway()` in `update_exporter_deb.py` and present the errors in a nicer format. Note that we still exit the Python script with an error, but since we ignore errors in `update-packages.sh` this should still ensure that failures when communicating with the push gateway don't fail the maintenance job.

Fixes #15

glrf reviewed (approved):
Code LGTM.

However shouldn't the script technically still return a non zero return code as it it failed and the bash script should handle it?

simu commented:
> However shouldn't the script technically still return a non zero return code as it it failed and the bash script should handle it?

Maybe that would be the better approach, I didn't think about it that deeply.

simu committed:
Ignore error return codes from `update_exporter_deb.py`

We ignore errors when trying to push the updated packages to Prometheus,
since a failure of that process shouldn't fail the maintenance job for a
node.

Fixes #15

simu committed:
Catch exceptions thrown by `pushadd_to_gateway()`

We catch exceptions thrown by `pushadd_to_gateway()` in
`update_exporter_deb.py` and present the errors in a nicer format.

Note that we still exit the Python script with an error, but since we
ignore errors in `update-packages.sh` this should still ensure that
failures when communicating with the push gateway don't fail the
maintenance job.
